src/clj_fdb/core.clj

```diff
@@ -120,3 +123,110 @@
+    & {:keys [t valfn]
+       :or {t (ftuple/from)
+            valfn bs/to-byte-array}}]
+   {:pre [(instance? Tuple t)]}
+   (set tc t v
+        :keyfn #(fsubspace/pack s %)
+        :valfn valfn)))
+
+(defn get-subspaced-key
+  "Takes the following:
+  - TransactionContext `tc`
+  - Subspace `s` which will be used to namespace the key
+
+  and returns byte-array `v` against key constructed using `s` and `t`.
+
+  Optionally, you can pass in `:t` and `:valfn` as follows:
+
+  - `:t` will be used along with `s` to construct key. `t` should of type `Tuple` class.
+  If `t` is not passed, empty `Tuple` will be used.
+  - `:valfn` should accept a byte-array and convert it as desired by the caller.
+  If no `:valfn` is provided, we return the byte-array as is."
+  ([^TransactionContext tc ^Subspace s
+    & {:keys [t valfn]
+       :or {t (ftuple/from)
+            valfn identity}}]
+   {:pre [(instance? Tuple t)]}
+   (get tc t
+        :keyfn #(fsubspace/pack s %)
+        :valfn valfn)))
+
+(defn clear-subspaced-key
+  "Takes the following:
+  - TransactionContext `tc`
+  - Subspace `s` which will used to namespace the range
+
+  and clears the Subspaced key from db. Returns nil.
+
+  Optionally, you can pass in `:t` as follows:
+
+  - `:t` will be used along with `s` to construct namespaced key. `t` should be of type `Tuple`.
+  If `t` is not passed, empty `Tuple` will be used."
+  [^TransactionContext tc ^Subspace s
+   & {:keys [t]
+      :or {t (ftuple/from)}}]
+  {:pre [(instance? Tuple t)]}
+  (clear tc t
+         :keyfn #(fsubspace/pack s %)))
+
+(defn get-subspaced-range
+  "Takes the following:
+  - TransactionContext `tc`
+  - Subspace `s` which will used to namespace the range
+
+  and returns a map of key/value pairs (byte-array->byte-array).
+
+  Optionally, you can pass in `:t`, `:keyfn` and `:valfn` as follows:
+
+  - `:t` will be used along with `s` to construct namespaced key. `t` should be of type `Tuple`.
+  If `t` is not passed, empty `Tuple` will be used.
+  - `:keyfn` can be passed to transform key to correct format.
+  If no `:keyfn` is provided, we return the byte-array as is.
+  - `:valfn` can be passed to tranform value to correct format.
+  If no `:valfn` is provided, we return the byte-array as is."
+  [^TransactionContext tc ^Subspace s
+   & {:keys [t keyfn valfn]
+      :or {t (ftuple/from)
+           keyfn identity
+           valfn identity}}]
+  {:pre [(instance? Tuple t)]}
+  (let [subspaced-range (fsubspace/range s t)]
+    (get-range tc subspaced-range
+               :keyfn keyfn
+               :valfn valfn)))
+
+(defn clear-subspaced-range
+  "Takes the following:
+  - TransactionContext `tc`
+  - Subspace `s` which will used to namespace the range
+
+  and clears the range from db. Returns nil.
+
+  Optionally, you can pass in `:t` as follows:
+
+  - `:t` will be used along with `s` to construct namespaced key. `t` should be of type `Tuple`.
+  If `t` is not passed, empty `Tuple` will be used."
```

Comment:
In this case, it makes sense to take `t` optionally. However, in the case that `t` is not provided, empty tuple should not be used (instead just call the single-arg version of `fsubspace/range`).